src/bsp_types/basic_bsp_structures.rs

```diff
@@ -224,3 +306,39 @@ pub struct BuildTargetCapabilities {
     /** This target can be debugged by the BSP server. */
     pub can_debug: bool,
 }
+
+
+impl BuildTargetCapabilities {
+    pub fn new() -> Self {
+        // set all to false
+        BuildTargetCapabilities::default()
+    }
+
+    pub fn set_all(&mut self) -> &mut Self {
+        self.can_compile = true;
+        self.can_test = true;
+        self.can_run = true;
+        self.can_debug = true;
+        self
+    }
+
+    pub fn set_compile(&mut self) -> &mut Self {
+        self.can_compile = true;
+        self
+    }
+
+    pub fn set_test(&mut self) -> &mut Self {
+        self.can_test = true;
+        self
+    }
+
+    pub fn set_run(&mut self) -> &mut Self {
+        self.can_run = true;
+        self
+    }
+
+    pub fn set_debug(&mut self) -> &mut Self {
+        self.can_debug = true;
+        self
+    }
+}
```

Comment:
czy te settery są potrzebne skoro wszystkie pola są publiczne?

Reply:
+ nawet jeśli byś je chciał zostawić to nie musisz zwracać self

Reply:
można zrobić pola prywatne ale nie wiem kto będzie chciał  z nich korzystać w przyszłości. Wszędzie mamy pola publiczne. Settery są zrobione w taki sposób, żeby można było je chainować. Patrząc na ich użycie zamiast pisać
{
capabilities.can_run = true
capabilities.can_compile = true
capabilities.can_debug = true
}

Można napisać

{
capabilities.set_debug().set_run().set_compile();
}